great_expectations/datasource/fluent/spark_filesystem_datasource.py

```diff
@@ -60,6 +61,7 @@ def add_csv_asset(
         header: bool = False,
         infer_schema: bool = False,
         order_by: Optional[SortersDefinition] = None,
+        batch_metadata: BatchMetadata | None = None,
```

Comment:
@billdirks I am curious how come the handling of metadata in the `filesystem` Spark datasource is different from the same location in its subclass, DBFS one?  Would we not want args and kwargs here as well?  Thanks.

Reply:
We actually don't want `args` and `kwargs` anywhere. The difference here is we have a `pyi` file and there is an incompatibility between mypy and pydantic we are working around. See my comment above `args` and `kwargs` in the DBFS file: https://github.com/great-expectations/great_expectations/pull/7534/files#diff-02c539ad8fa0586577ecb10503b4d28ac53f6b5180fe320de8c42f629dfa21fcR45

This is the pattern @NathanFarmer and @Kilo59 decided upon when they did the initial change in Pandas datasources.

Reply:
I think I can actually get rid of `args` and `kwargs` below since we don't have any additional `kwargs` like we do in the pandas case.